src/main/resources/config/application.yml

```diff
@@ -0,0 +1,67 @@
+# All env variables that are used in one place
+# Exception: TERRA_ENV (which sets the active profile), which is parsed before this is.
+env:
+  db:
+    host: jdbc:postgresql://${DATABASE_HOSTNAME:127.0.0.1}:5432
+    init: ${INIT_DB:false}
+    stairway:
+      name: ${STAIRWAY_DATABASE_NAME:stairwaylib}
+      pass: ${STAIRWAY_DATABASE_USER_PASSWORD:stairwaypwd}
+      user: ${STAIRWAY_DATABASE_USER:stairwayuser}
+    ws:
+      name: ${STAIRWAY_DATABASE_NAME:wm}
+      pass: ${STAIRWAY_DATABASE_USER_PASSWORD:wmpwd}
+      user: ${STAIRWAY_DATABASE_USER:wmuser}
+  gcp-sa-path: ${CLOUD_TRACE_SA_PATH:rendered/service-account.json}
+
+logging.pattern.level: '%X{requestId} %5p'
+
+server:
+  compression:
+    enabled: true
+    mime-types: text/css,application/javascript
+  port: 8080
+
+spring:
+  cloud.gcp:
+    credentials.location: file:${env.gcp-sa-path}
+    trace.enabled: true
+  profiles.active: ${TERRA_ENV:dev} # This is evaluated before the "env" section above
+  resources:
+    cache:
+      cachecontrol:
+        max-age: 0
+        must-revalidate: true
+    static-locations: classpath:/api/
+  sleuth.sampler.probability: 0.1
+
+workspace:
+  datarepo.instances:
+    terra-alpha: https://data.alpha.envs-terra.bio
+    terra-dev: https://jade.datarepo-dev.broadinstitute.org
+    terra-prod: https://jade-terra.datarepo-prod.broadinstitute.org
+    terra-staging: https://data.staging.envs-terra.bio
+
+  db:
+    stairway:
+      forceClean: true
+      migrateUpgrade: ${env.db.init}
+      password: ${env.db.stairway.pass}
+      uri: ${env.db.host}/${env.db.stairway.name}
+      username: ${env.db.stairway.user}
+    workspace:
```

Comment:
[nit / opt] The config path here, `workspace.db.workspace` feels a bit awkward due to repetition.

What I did for Janitor was to nest the Janitor-specific DB config directly under the Janitor namespace (e.g. `janitor.db`), and put the Stairway DB stuff under `janitor.stairway.db`. It flips around the hierarchy a bit but felt reasonable enough to me. ([code ref](https://github.com/DataBiosphere/terra-resource-janitor/blob/master/src/main/resources/application.yml#L2))